h3py/h3core.pyx

```diff
@@ -0,0 +1,274 @@
+cimport h3py.util as u
+from h3py.util import H3ValueError
+from h3py.util cimport create_ptr, create_mv
+
+from cpython cimport bool
+
+from h3py.libh3 cimport H3int
+cimport h3py.libh3 as h3c
+
+
+from h3py.geo import (
+    geo_to_h3,
+    h3_to_geo,
+    polyfill,
+    cell_boundary,
+    edge_boundary
+)
+
+# todo: should we rename all `hex` things to `cell` if a pentagon is a possibility?
+# what's the difference between a `cell` and a `base cell`?
+
+
+# bool is a python type, so we don't need the except clause
+cpdef bool is_cell(H3int h):
+    """Validates an `h3_address`
+
+    :returns: boolean
+    """
+    return h3c.h3IsValid(h) == 1
+
+cpdef bool is_pentagon(H3int h):
+    return h3c.h3IsPentagon(h) == 1
+
+cpdef int base_cell(H3int h) except -1:
+    # todo: `get_base_cell` a better name?
+    u.check_addr(h)
+
+    return h3c.h3GetBaseCell(h)
+
+
+cpdef int resolution(H3int h) except -1:
+    """Returns the resolution of an `h3_address`
+    0--15
+    """
+    u.check_addr(h)
+
+    return h3c.h3GetResolution(h)
+
+
+cpdef H3int parent(H3int h, res=None) except 1:
+    u.check_addr(h)
+
+    if res is None:
+        res = resolution(h) - 1
+    # todo: actually, do we want to raise an error if there are no children, or just return an empty set?
+    u.check_res(res)
+
+    return h3c.h3ToParent(h, res)
+
+
+cpdef int distance(H3int h1, H3int h2) except -1:
+    """ compute the hex-distance between two hexagons
+    """
+    u.check_addr(h1)
+    u.check_addr(h2)
+
+    d = h3c.h3Distance(h1,h2)
+
+    return d
+
+cpdef H3int[:] disk(H3int h, int k):
```

Comment:
I'm a little concerned we're going to proliferate more terms, when we probably have too many terms for kRing already (I'd like to reduce the number of unique terms we use for that in H3 4.x)

Reply:
Agreed. This was just an experiment to try to fix the confusion around `k_ring` and `hex_ring`.

Reply:
Also, the `old` API doesn't expose this term to the users.

Reply:
I've updated the naming to match the current python implementation.